core/deployment/src/test/java/io/quarkus/deployment/pkg/steps/JarResultBuildStepTest.java

```diff
@@ -31,4 +67,40 @@ void should_unsign_jar_when_filtered(@TempDir Path tempDir) throws Exception {
         }
     }
 
+    private static KeyStore.PrivateKeyEntry createPrivateKeyEntry()
+            throws NoSuchAlgorithmException, CertificateException, OperatorCreationException, CertIOException {
+        KeyPairGenerator ky = KeyPairGenerator.getInstance("RSA");
+        ky.initialize(2048);
+        KeyPair keyPair = ky.generateKeyPair();
+        Certificate[] chain = { createCertificate(keyPair, "cn=Unknown") };
+        KeyStore.PrivateKeyEntry keyEntry = new KeyStore.PrivateKeyEntry(keyPair.getPrivate(), chain);
+        return keyEntry;
+    }
+
+    private static Certificate createCertificate(KeyPair keyPair, String subjectDN)
+            throws OperatorCreationException, CertificateException, CertIOException {
+        Provider bcProvider = new BouncyCastleProvider();
+        Security.addProvider(bcProvider);
+        long now = System.currentTimeMillis();
+        Date startDate = new Date(now);
+        X500Name dnName = new X500Name(subjectDN);
+        BigInteger certSerialNumber = new BigInteger(Long.toString(now));
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(startDate);
```

Comment:
AFAIK a Calendar is already initialized with the current time, why are you setting the time here?